File: src/java/org/javaturk/dp/ch04/abstractFactory/university/sol2/FactoryForUniversityB.java
package org.javaturk.dp.ch04.abstractFactory.university.sol2;

public interface FactoryForUniversityB {
	
	Course createCourse();
	Professor createProfessor();
	Schedule createSchedule();
	Syllabus createSyllabus();
	VirtualClassroom createVirtualClassroom();
}

File: src/typescript/com/gokselkucuksahin/dp/creational-patterns/abstract-factory/university/sol2/factory-for-university-b.ts
import {VirtualClassroom} from "./virtual-classroom";
import {Course} from "../sol1/course";
import {Professor} from "../sol1/professor";
import {Schedule} from "../sol1/schedule";
import {Syllabus} from "../sol1/syllabus";

export interface FactoryForUniversityB {
  createCourse(): Course;

  createProfessor(): Professor;

  createSchedule(): Schedule;

  createSyllabus(): Syllabus;

  createVirtualClassromm(): VirtualClassroom;
}